addons/stock_account/tests/test_account_move.py

```diff
@@ -298,3 +298,45 @@ def test_product_valuation_method_change_to_automated_negative_on_hand_qty(self)
             [stock_valuation_line, output_line],
             [expected_valuation_line, expected_output_line]
         )
+
+    def test_stock_account_move_automated_not_standard_with_branch_company(self):
+        """
+        Test that the validation of a stock picking does not fail `_check_company`
+        at the creation of the account move with sub company
+        """
+        main_company = self.env.company
+        branch_a = self.env['res.company'].create({
+            'name': 'Branch A',
+            'country_id': main_company.country_id.id,
+            'parent_id': main_company.id,
+        })
+
+        self.env.user.write({'company_ids': (main_company + branch_a).ids, 'company_id': branch_a.id})
+
+        self.cr.precommit.run()
+
+        categ = self.auto_categ.copy({'property_cost_method': 'average', 'property_valuation': 'real_time'})
```

Comment:
Let's avoid a record creation
```suggestion
        self.auto_categ.write({'property_cost_method': 'average', 'property_valuation': 'real_time'})
```